### PR TITLE
kvstore change empty to nil

### PR DESCRIPTION
**Pull request by xgess**
* update the types to the most recent keybase client release
* support new change to kvstore entryValue behavior (`None` instead of `""` when unset)

depends on https://github.com/keybase/client/pull/23884

suggest:
1. review this by commit
2. this is safe to merge and publish before the keybase client goes out because it will just pass through whatever the client says.

### CODE REVIEW

**Comment by xgess** [comment]
@keybase/picnicsquad

File: examples/5_secret_storage.py [hunk 1]
```diff
@@ -260,10 +266,15 @@ async def remove(
                 most recent get result if applicable)
         """
         res = await self.lookup(team, tool)
-        if res.entry_value == "":
+        if res.entry_value is not None:
```
Comment (mmou) [inline examples/5_secret_storage.py:269]:
```suggestion
        if res.entry_value is None:
```

File: examples/5_secret_storage.py [hunk 2]
```diff
@@ -451,7 +462,7 @@ def noop_handler(*args, **kwargs):
     rental = RentalBotClient(bot)
 
     print("...basic rental actions...")
-    await basic_rental_users(bot, rental, team)
+    await basic_rental_users(rental, team)
 
     print("...multiple users try to reserve...")
     await concurrent_rental_users(bot, rental, team)
```
Comment (mmou) [inline examples/5_secret_storage.py:468]: i suppose we could remove `bot` from this function args too?

File: examples/5_secret_storage.py [hunk 3]
```diff
@@ -102,15 +104,15 @@ async def delete(
         return res
 
     async def get(
-        self, team: str, namespace: str, entry_key: str
```
Comment (mmou) [inline examples/5_secret_storage.py:105]: thanks :pray:

File: examples/5_secret_storage.py [hunk 4]
```diff
@@ -18,6 +18,8 @@
 #
 # Here we've stored the HMAC secret and other entries in the team's kvstore;
 # you could also store the entries in the bot's own kvstore (the default team).
+#
+# TODO: this is currently not working exactly right. sorry.
```
Comment (mmou) [inline examples/5_secret_storage.py:22]: should we file a ticket w/ more details?

File: pykeybasebot/kvstore_client.py [hunk 5]
```diff
@@ -98,10 +98,10 @@ async def list_entrykeys(
         return keybase1.KVListEntryResult.from_dict(res)
 
     def is_deleted(self, res: keybase1.KVGetResult) -> bool:
-        return res.revision > 0 and res.entry_value == ""
+        return res.revision > 0 and res.entry_value is None
```
Comment (mmou) [inline pykeybasebot/kvstore_client.py:101]: i noticed the json bot lib PR keeps this backwards compatible by also checking whether res.entry_value  == "". should we be consistent across the bot libs?

File: examples/5_secret_storage.py (hunk 1 above)
Comment (xgess) [reply examples/5_secret_storage.py:269]: oops! fixed.

File: examples/5_secret_storage.py (hunk 2 above)
Comment (xgess) [reply examples/5_secret_storage.py:468]: done

File: examples/5_secret_storage.py (hunk 4 above)
Comment (xgess) [reply examples/5_secret_storage.py:22]: sure. i'll do it.

File: pykeybasebot/kvstore_client.py (hunk 5 above)
Comment (xgess) [reply pykeybasebot/kvstore_client.py:101]: yup. fixed.

**Pull request merged**